pyerrors/fits.py

```diff
@@ -656,6 +690,189 @@ def chisqfunc_compact(d):
     return output
 
 
+def _combined_fit(x, y, func, silent=False, **kwargs):
+
+    if kwargs.get('correlated_fit') is True:
+        raise Exception("Correlated fit has not been implemented yet")
+
+    output = Fit_result()
+    output.fit_function = func
+
+    if kwargs.get('num_grad') is True:
+        jacobian = num_jacobian
+        hessian = num_hessian
+    else:
+        jacobian = auto_jacobian
+        hessian = auto_hessian
+
+    x_all = np.concatenate([np.array(o) for o in x.values()])
+    y_all = np.concatenate([np.array(o) for o in y.values()])
```

Comment:
You don't have to rely on the order of the dictionary here. Use the keys of one of the dictionaries to ensure that corresponding (key, value) pairs of the three dictionaries are properly matched.
``` python
keylist = list(x.keys())
x_all = np.concatenate([np.array(x[k]) for k in keylist])
y_all = np.concatenate([np.array(y[k]) for k in keylist])
```
You should also use this to check for consistency of the three dictionaries (do they contain the same keys? Do the (x, y)-pairs match for all keys? ...)